lib/iris/experimental/regrid.py

```diff
@@ -879,29 +942,51 @@ def regrid_weighted_curvilinear_to_rectilinear(src_cube, weights, grid_cube):
     def _src_align_and_flatten(coord):
         # Return a flattened, unmasked copy of a coordinate's points array that
         # will align with a flattened version of the source cube's data.
+        #
+        # PP-TODO: Should work with any cube dimensions for X and Y coords.
+        #  Probably needs fixing anyway?
+        #
         points = coord.points
         if src_cube.coord_dims(coord) == (1, 0):
             points = points.T
         if points.shape != src_cube.shape:
-            msg = 'The shape of the points array of !r is not compatible ' \
-                'with the shape of !r.'.format(coord.name(), src_cube.name())
-            raise ValueError(msg)
+            msg = 'The shape of the points array of {!r} is not compatible ' \
+                'with the shape of {!r}.'
+            raise ValueError(msg.format(coord.name(), src_cube.name()))
         return np.asarray(points.flatten())
 
     # Align and flatten the coordinate points of the source space.
     sx_points = _src_align_and_flatten(sx)
     sy_points = _src_align_and_flatten(sy)
 
-    # Match the source cube x coordinate range to the target grid
-    # cube x coordinate range.
-    min_sx, min_tx = np.min(sx.points), np.min(tx.points)
+    # Transform source X and Y points into the target coord-system, if needed.
+    if sx.coord_system != tx.coord_system:
+        src_crs = sx.coord_system.as_cartopy_projection()
+        tgt_crs = tx.coord_system.as_cartopy_projection()
+        sx_points, sy_points = _transform_xy_arrays(
+            src_crs, sx_points, sy_points, tgt_crs)
+    #
+    # PP-TODO: how does this work with scaled units ??
```

Comment:
Very much a problem for another day I think.

I consulted @marqh on this, and he thought it quite possible that we just haven't considered this in other places in the Iris code where we transform points with Cartopy.
Cartopy itself has no concept of units :  I think it basically assumes either degrees or metres for everything, on the principle that 'metres' is a defined concept on a map surface.  The problem is, it publishes no obvious record of **which** projections are in degrees.
It might be a good idea for every Cartopy projection to somehow specify the coordinate units for each projection, then we could close the loop on this uncertainty.